edgelet/aziot-edged/src/watchdog.rs

```diff
@@ -134,6 +140,34 @@ async fn create_and_start_agent(
     let mut env = agent_env(gen_id, settings, device_info);
     agent_spec.env_mut().append(&mut env);
 
+    let pinfo = if let Some(path) = settings.product_info() {
+        Ok(
+            crate::product_info::ProductInfo::try_load(path).map_err(|err| {
+                EdgedError::from_err(
+                    format!("malformed product info at path {}", path.display()),
+                    err,
+                )
+            })?,
+        )
+    } else {
+        crate::product_info::ProductInfo::from_system()
+    };
+
+    match pinfo {
+        Ok(pinfo) => {
+            let prev = agent_spec
+                .env_mut()
+                .insert("IOTEDGE_PRODUCTINFO".to_owned(), pinfo.to_string());
+
+            if prev.is_some() {
+                log::warn!("overrode old IOTEDGE_PRODUCTINFO environment variable");
```

Comment:
This can only ever happen if someone set IOTEDGE_PRODUCTINFO in agent config. There's no use case for that, so I don't think we need to consider this.